src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -997,6 +1006,11 @@ internal bool MethodGroupIsCompatibleWithDelegate(BoundExpression receiverOpt, b
                 return sourceConstantValue;
             }
 
+            if (source.HasAnyErrors)
```

Comment:
What is affected by moving this HasAnyErrors down?

Reply:
The `int i = default;` scenario in C# 7.1 (ie before the `default` literal feature) would not give `default` a constant value (because there is an error on `default`). But we can give it its constant value (instead of `null`) by moving this check down.

---
In reply to: [313659911](https://github.com/dotnet/roslyn/pull/37596#discussion_r313659911) [](ancestors = 313659911)